src/lib.rs

```diff
@@ -381,10 +381,22 @@ pub fn parse_path(path: &PathBuf, config: ParseConfig) -> anyhow::Result<ParseOu
                 .map_or(
                     format!("unknown_{lineno}"),
                     |CompileId {
+                         compiled_autograd_id,
                          frame_id,
                          frame_compile_id,
                          attempt,
-                     }| { format!("{frame_id}_{frame_compile_id}_{attempt}") },
+                     }| {
+                        let frame_id_str = frame_id.map_or("-".to_string(), |v| v.to_string());
+                        let frame_compile_id_str =
+                            frame_compile_id.map_or("-".to_string(), |v| v.to_string());
+                        let attempt_str = attempt.map_or("-".to_string(), |v| v.to_string());
+
+                        if let Some(ca_id) = compiled_autograd_id {
+                            format!("{ca_id}_{frame_id_str}_{frame_compile_id_str}_{attempt_str}")
```

Comment:
Since the file naming here isn't load bearing, I wonder if we should do something wordier like "compiled_autograd_{ca_id}_..." so it's obvious that compiled autograd is involved here. Well, I guess the filename doesn't matter that much.

Reply:
we could use abbreviations like ca{id}_... to keep things compact and to stick under the 255 char limit for windows